src/main/java/com/auth0/client/auth/AuthAPI.java

```diff
@@ -553,7 +558,53 @@ public AuthRequest login(String emailOrUsername, char[] password, String realm)
         request.addParameter(KEY_GRANT_TYPE, "http://auth0.com/oauth/grant-type/password-realm");
         request.addParameter(KEY_USERNAME, emailOrUsername);
         request.addParameter(KEY_PASSWORD, password);
-        request.addParameter("realm", realm);
+        request.addParameter(KEY_REALM, realm);
+        return request;
+    }
+
+    /**
+     * Creates a login request using the Passwordless grant type.
+     *
+     * <pre>
+     * {@code
+     * AuthAPI auth = new AuthAPI("me.auth0.com", "B3c6RYhk1v9SbIJcRIOwu62gIUGsnze", "2679NfkaBn62e6w5E8zNEzjr-yWfkaBne");
+     * try {
+     *      TokenHolder result = auth.login("user@domain.com", PasswordlessRealmType.EMAIL, new char[]{'c','o','d','e'})
+     *          .execute();
+     * } catch (Auth0Exception e) {
+     *      // Something happened
+     * }
+     * }
+     * </pre>
+     *
+     * @param emailOrPhone The email or phone number of the user.
+     * @param realm The realm to use.
+     * @param otp The one-time password used to authenticate using Passwordless connections
+     *
+     * @return A request to configure and execute
+     *
+     * @see <a href="https://auth0.com/docs/connections/passwordless/reference/relevant-api-endpoints">Using Passwordless APIs</a>
+     * @see com.auth0.client.auth.AuthAPI#startPasswordlessEmailFlow(String, PasswordlessEmailType)
+     * @see com.auth0.client.auth.AuthAPI#startPasswordlessSmsFlow(String)
+     */
+    public AuthRequest login(String emailOrPhone, PasswordlessRealmType realm, char[] otp) {
```

Comment:
We overload the `login` method several times in `AuthAPI` , so I followed that pattern here. That said, I wonder if it makes sense, or if it should be something like `exchangePasswordlessOtp` or something along those lines?

Reply:
It's a good question. I lean towards the latter because I find it easier to type and use, rather than typing "login" and scrolling through the N suggested alternatives. Also, I know we plan to add MFA as well, so that other one could look like `exchangeMfaOtp` for example. 

What's the `realm` parameter used for here? Can we assume that value instead of having to receive it? Since you have an object and then just call `getRealm` on it maybe it's simpler to receive a string. Unless you abstracted this because you are using it [on 3 or more places](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). 

About the use of char arrays for passwords. I remember someone raised this in the past. This is a temporary code that shouldn't last more than 20 seconds. Do we still want to use char arrays? There's no wrong answer.

Reply:
I agree with your points regarding the name - I will change it to `exchangePasswordlessOtp`. 

> What's the realm parameter used for here? Can we assume that value instead of having to receive it? Since you have an object and then just call getRealm on it maybe it's simpler to receive a string.

The realm is used to specify the delivery type of the code sent to the user: "Use sms or email (should be the same as POST /passwordless/start)" ([docs](https://auth0.com/docs/api/authentication#authenticate-user)). Unless we create separate methods for email/sms, we cannot infer it. As to whether it should just accept a simple string, that is a viable option, and the way that other methods in this SDK work. An enum provides some additional protections as they are fixed constants, so a developer could not (for example), specify an invalid value. That said, it does also mean that if other valid realm types are added, the enum would require updating. 

> This is a temporary code that shouldn't last more than 20 seconds. Do we still want to use char arrays? There's no wrong answer.

I thought about this for the same reason (short duration code). For that reason using a String does not present significant risk, but chose a char array to err on the side of safety.

Reply:
re char array / string, I'm fine either way. re realm, do notice that passwordless connections other than the default "SMS" and "email" **can** be created via the management API. So we should support any value here.